tests/src/test/scala/common/WskTestHelpers.scala

```diff
@@ -31,6 +31,55 @@ import spray.json._
 import java.time.Instant
 
 /**
+ * An arbitrary response of a whisk action. Includes the result as a JsObject as the
+ * structure of "result" is not defined.
+ */
+case class activationResponse(result: Option[JsObject], status: String, success: Boolean)
+
+object activationResponse extends DefaultJsonProtocol {
+    implicit val serdes = jsonFormat3(activationResponse.apply)
+}
+
+/**
+ * Activation record as it is returned by the CLI.
+ */
+ case class activationResult (
+        activationId: String,
+        logs: Option[List[String]],
+        response: activationResponse,
+        start: Instant,
+        end: Option[Instant], // This field is optional in responses of REST calls.
+        duration: Option[Long], // This field is optional in responses of REST calls.
```

Comment:
When we try to access the information of activation, sometimes the activation only has start available, no end or duration.
In this situation, CLI will fill in end and duration with some default values, while REST response will have them missing. When we to convertTo[activationResponse], there is no problem for CLI response, cos they will have values anyway. However, if we convert the http response into activationResponse, we need to tolerate end and duration missing. That is why they are optional.